File: C/70-climbing-stairs/main.go
package main

import "fmt"

func climbStairs(n int) int {
	if n <= 2 {
		return n
	}
	b := make([]int, n)
	b[0] = 1
	b[1] = 2
	for i := 2; i < n; i++ {
		b[i] = b[i-1] + b[i-2]
	}
	return b[n-1]
}

func main() {
	fmt.Println("a")
}

File: C/70-climbing-stairs/main.rs
struct Solution {}

impl Solution {
  pub fn climb_stairs(n: i32) -> i32 {
    let mut dp = vec![0; n as usize + 1];
    dp[0] = 1;
    (0..n as usize).for_each(|i| {
      dp[i + 1] += dp[i];
      if i + 2 <= n as usize {
        dp[i + 2] += dp[i];
      }
    });
    dp[n as usize]
  }
}
